connect/proxy/listener.go

```diff
@@ -171,7 +175,6 @@ func (l *Listener) handleConn(src net.Conn) {
 
 	// Make sure Close() waits for this conn to be cleaned up. Note defer is
 	// before conn.Close() so runs after defer conn.Close().
-	l.connWG.Add(1)
 	defer l.connWG.Done()
```

Comment:
I think this needs to be the very first line of this function so it pairs well with the WaitGroup.Add